python/src/aac/lang/language_context.py

```diff
@@ -219,12 +222,11 @@ def update_definition_in_context(self, definition: Definition):
             definition (Definition): The Definition to update in the context.
         """
 
-        if definition.uid in self.definitions_dictionary:
-            old_definition = self.definitions_dictionary.get(definition.uid)
-
+        old_definition = self.definitions_dictionary.get(definition.uid)
+        if old_definition and self._is_definition_editable(definition):
             self.remove_definition_from_context(old_definition)
             self.add_definition_to_context(definition)
-        else:
+        elif definition.uid not in self.definitions_dictionary:
             definitions_in_context = self.get_defined_types()
             logging.error(
                 f"Definition not present in context, can't be updated. '{definition.name}' not in '{definitions_in_context}'"
```

Comment:
I can't leave a comment on the other lines but this also applies to lines 185-187. I definitely think we should log an error but (at least for the Rest API) only logging an error message doesn't give the user feedback that what they tried to do failed unless they happen to look at the logs (but I wouldn't assume someone would do so when the API indicates that it succeeded). This is probably something we need to consider making a bug to handle later because actually raising an error may cause issues elsewhere, but I think we should be aware of the current limitation.

Reply:
This should be covered by the lang error raised when this happens.